commit fc669f7af3382f0886beb80e08eaac0d17116029
Author: 12101111 <w12101111@gmail.com>
Date:   Tue Mar 30 10:27:04 2021 +0800

    Revert "util: Remove the L_AUTO_CLEANUP_VAR macro"
    
    This reverts commit 92369d27b99688766a8b2c96760ea20cf6c8ced4.

diff --git a/ell/util.h b/ell/util.h
index 9400657..745740c 100644
--- a/ell/util.h
+++ b/ell/util.h
@@ -226,6 +226,9 @@ static inline void l_put_be64(uint64_t val, void *ptr)
 	L_PUT_UNALIGNED(L_CPU_TO_BE64(val), (uint64_t *) ptr);
 }
 
+#define L_AUTO_CLEANUP_VAR(vartype,varname,destroy) \
+	vartype varname __attribute__((cleanup(destroy)))
+
 #define L_AUTO_FREE_VAR(vartype,varname) \
 	vartype varname __attribute__((cleanup(auto_free)))
 
